### PR TITLE
[MOB-9996] User update rearchitecting

### PR DESCRIPTION
**Pull request by evantk91**
## 🔹 Jira Ticket(s)

* [MOB-9996](https://iterable.atlassian.net/browse/MOB-9996)

## ✏️ Description

> This pull request separates the user update object in local storage to prevent user updates from being lost. This also simplifies a lot of the event processing code.

[MOB-9996]: https://iterable.atlassian.net/browse/MOB-9996?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: swift-sdk/Internal/AnonymousUserManager.swift [hunk 1]
```diff
@@ -144,37 +126,32 @@ public class AnonymousUserManager: AnonymousUserManagerProtocol {
...
+                        IterableAPI.implementation?.track(eventData[JsonKey.eventName] as? String ?? "", withBody: eventData)
                         break
                     case EventType.purchase:
                         var total = NSNumber(value: 0)
                         if let _total = NumberFormatter().number(from: eventData[JsonKey.Commerce.total] as! String) {
                             total = _total
                         }
                         
-                        IterableAPI.implementation?.trackPurchase(total, items: convertCommerceItems(from: eventData[JsonKey.Commerce.items] as! [[AnyHashable: Any]]), dataFields: eventData[JsonKey.dataFields] as? [AnyHashable : Any], createdAt: eventData[JsonKey.Body.createdAt] as? Int ?? 0, onSuccess: {result in
-                            successfulSyncedData.append(eventData[JsonKey.eventTimeStamp] as? Int ?? 0)
```
Comment (Ayyanchira) [inline swift-sdk/Internal/AnonymousUserManager.swift:167]: What was this callback supposed to do?

**Review by Ayyanchira — APPROVED** [review]
LGTM :)

File: swift-sdk/Internal/AnonymousUserManager.swift (hunk 1 above)
Comment (evantk91) [reply swift-sdk/Internal/AnonymousUserManager.swift:167]: We determine nothing haha. It is just storing synced data locally.

**Review by evantk91 — COMMENTED** [review]
I am making some modifications as the user updates were not being updated properly.

**Pull request merged**